File: 프로그래머스/lv2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    people = people.sort((a,b) => b - a);
    let count = 0;
     for(let i = 0; i < people.length; i++) {
       if(people[i] + people[people.length - 1] > limit) {
           count++;
        } else {
            people.pop();
            count++;
        }
    } 
    return count;
}

File: 프로그래머스/lv2/42885. 구명보트/구명보트.java
import java.util.*;
class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        Arrays.sort(people);
        Stack<Integer> stack = new Stack<>();
        for(int i = people.length - 1; i >= 0; i--) {
            stack.push(people[i]);
        }
        int index = 0;
        while(index < stack.size()) {
           if(stack.size() >= 2) {
             if(stack.get(index) + stack.peek() > limit) {
                answer++;
            } else {
                stack.pop();
                answer++;;
            }    
           } else {
              stack.pop(); 
              answer++;;
           }
           index++;
        }
        return answer;
    }
}